ui/components/app/gas-customization/advanced-gas-inputs/advanced-gas-inputs.component.js

```diff
@@ -52,22 +52,32 @@ export default class AdvancedGasInputs extends Component {
     }
   }
 
-  onChangeGasLimit = (e) => {
-    this.setState({ gasLimit: e.target.value });
-    this.changeGasLimit({ target: { value: e.target.value } });
+  onChangeGasLimit = ({ target: { value } }) => {
+    const newGasLimit = Number(value);
+    if (isNaN(newGasLimit) || newGasLimit < 0) {
+      this.setState({ gasLimit: this.state.gasLimit });
+      return;
+    }
+    this.setState({ gasLimit: newGasLimit });
+    this.changeGasLimit(newGasLimit);
   };
 
-  changeGasLimit = (e) => {
-    this.props.updateCustomGasLimit(Number(e.target.value));
+  changeGasLimit = (newGasLimit) => {
+    this.props.updateCustomGasLimit(newGasLimit);
   };
 
-  onChangeGasPrice = (e) => {
-    this.setState({ gasPrice: e.target.value });
-    this.changeGasPrice({ target: { value: e.target.value } });
+  onChangeGasPrice = ({ target: { value } }) => {
+    const newGasPrice = Number(value);
+    if (isNaN(newGasPrice) || newGasPrice < 0) {
+      this.setState({ gasPrice: this.state.gasPrice });
```

Comment:
Not sure why is this setState needed `this.setState({ gasPrice: this.state.gasPrice });` ?

Reply:
ideally this would be an `event.preventDefault()` call, but this will not work for `onChange` events, this is to restore the input to the last valid value. See chat for more details on this particular PR :)

Reply:
So just to be clear - this is being done to overwrite the value in the actual text input HTML element? The state isn't changing here, but setting state here has the side-effect of triggering another render cycle that writes this back to the element.

Reply:
correct yes